src/main/java/games/strategy/triplea/ai/weakAI/WeakAI.java

```diff
@@ -864,6 +863,7 @@ public void purchase(final boolean purchaseForBid, final int PUsToSpend, final I
       //
       // if capitol is super safe, we don't have to do this. and if capitol is under siege, we should repair enough to
       // place all our units here
+      int maxUnits = (totPU - 1) / minimumUnitPrice;
```

Comment:
Both `totPU` and `minimumUnitPrice` are `final`.  So there is no risk that they are modified between old line 834 and new line 866.  Thus, `maxUnits` will have the same initial value after being moved.